src/main/kotlin/com/sipios/springsearch/strategies/EnumStrategy.kt

```diff
@@ -0,0 +1,28 @@
+package com.sipios.springsearch.strategies
+
+import com.sipios.springsearch.SearchOperation
+import javax.persistence.criteria.CriteriaBuilder
+import javax.persistence.criteria.Path
+import javax.persistence.criteria.Predicate
+import kotlin.reflect.KClass
+
+class EnumStrategy : ParsingStrategy {
+    override fun buildPredicate(
+        builder: CriteriaBuilder,
+        path: Path<*>,
+        fieldName: String,
+        ops: SearchOperation?,
+        value: Any?
+    ): Predicate? {
+        return when (ops) {
+            SearchOperation.EQUALS -> builder.equal(path.get<Any>(fieldName), value)
+            SearchOperation.NOT_EQUALS -> builder.notEqual(path.get<Any>(fieldName), value)
+            else -> super.buildPredicate(builder, path, fieldName, ops, value)
+        }
+    }
+
+    override fun parse(value: String?, fieldClass: KClass<out Any>): Any? {
+        val enumClz = Class.forName(fieldClass.qualifiedName).enumConstants as Array<Enum<*>>
+        return enumClz.first { it.name == value }
```

Comment:
I think we could use an easier way to return the Enum corresponding to the value being searched. 
Did you take a look at the following [Baeldung article](https://www.baeldung.com/java-string-to-enum) ? 
It could be interesting to try this method :)

Reply:
Yes I saw
We need to do this in reflection and create a instance of this based on the class.

Reply:
return Class.forName(fieldClass.qualifiedName).methods.first { it.name == "valueOf" }.invoke(null, value)

Reply:
Good for me the second one :)